forge-game/src/main/java/forge/game/cost/CostPutCounter.java

```diff
@@ -141,16 +141,16 @@ public final void refund(final Card source) {
     public final boolean canPay(final SpellAbility ability, final Player payer, final boolean effect) {
         final Card source = ability.getHostCard();
         if (this.payCostFromSource()) {
-            return source.canReceiveCounters(this.counter);
-        } else {
-            // 3 Cards have Put a -1/-1 Counter on a Creature you control.
-            List<Card> typeList = CardLists.getValidCards(source.getGame().getCardsIn(ZoneType.Battlefield),
-                    this.getType().split(";"), payer, source, ability);
+            return source.isInPlay() && source.canReceiveCounters(this.counter);
```

Comment:
is this check good enough for when source gets blinked?

Reply:
Yea, the source will then have the zone it moves to (and keep that even if it returns)

Reply:
so in case the creature gets blinked, are you able to pay the cost or not?

Reply:
You'll get a second trigger for which you could pay

But the old one will just create a token, because the object on which the counter would be put no longer exists